### PR TITLE
improve Speak & Speak and wait Bricks

### PR DESCRIPTION
**Pull request by ndrnour**
see attached Pics

### CODE REVIEW

**Comment by ndrnour** [comment]
please run espresso tests

**Comment by ndrnour** [comment]
please run espresso tests

**Comment by ndrnour** [comment]
please run espresso tests

File: catroid/src/main/java/org/catrobat/catroid/content/bricks/SpeakBrickTools.java [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+import org.catrobat.catroid.R;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+
+import static org.catrobat.catroid.common.Constants.AVAILABLE_LOCALES_TTS;
+
+public class SpeakBrickTools {
```
Comment (84n4n4) [inline catroid/src/main/java/org/catrobat/catroid/content/bricks/SpeakBrickTools.java]: please create a base class for both speak bricks and move all those methods there

File: catroid/src/main/java/org/catrobat/catroid/content/bricks/SpeakBrickTools.java [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+
+	public void setLocale(Locale mlocale) {
+		locale = mlocale;
+	}
+
+	public static Locale getLocale() {
+		return locale;
+	}
+
+	static void setSpeakSpinner(Context context, View view, boolean isPrototypeView) {
```
Comment (84n4n4) [inline catroid/src/main/java/org/catrobat/catroid/content/bricks/SpeakBrickTools.java]: please split in two methods, eg. setupSpinner and setItemSelectedListener and remove the isPrototypeView parameter. either call one or both methods depending on what you need.

File: catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/content/brick/app/SpeakBrickTest.java [hunk 3]
```diff
@@ -63,6 +69,15 @@ public void speakBrickTest() {
 
 		onBrickAtPosition(0).checkShowsText(R.string.brick_when_started);
 		onBrickAtPosition(brickPosition).checkShowsText(R.string.brick_speak);
+		onBrickAtPosition(brickPosition).checkShowsText(R.string.brick_speak_in_language);
+
+		onData(anything()).inAdapterView(withId(R.id.brick_speak_spinner)).atPosition(0).check(matches(withText(R
```
Comment (84n4n4) [inline catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/content/brick/app/SpeakBrickTest.java]: create a wrapper for the locale spinner

File: catroid/src/main/java/org/catrobat/catroid/stage/PreStageActivity.java [hunk 4]
```diff
@@ -568,8 +568,24 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
 					textToSpeech = new TextToSpeech(getApplicationContext(), new OnInitListener() {
 						@Override
 						public void onInit(int status) {
-							onUtteranceCompletedListenerContainer = new OnUtteranceCompletedListenerContainer();
-							textToSpeech.setOnUtteranceCompletedListener(onUtteranceCompletedListenerContainer);
+							utteranceProgressListener = new UtteranceProgressListener() {
```
Comment (84n4n4) [inline catroid/src/main/java/org/catrobat/catroid/stage/PreStageActivity.java]: please put this in a separate method.

**Comment by ndrnour** [comment]
please run espresso tests

File: catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/content/brick/app/SpeakBrickTest.java (hunk 3 above)
Comment (ndrnour) [reply catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/content/brick/app/SpeakBrickTest.java]: I've no idea how to create a wrapper for a such a spinner with "List<locale> as arrayAdapter" and "Strings(R.string.device_language + displayNames of the locales) as views".
by the way, as I did above works fine!

**Comment by ndrnour** [comment]
please run espresso tests

**Comment by ndrnour** [comment]
`org.catrobat.catroid.uiespresso.content.brick.stage.SayForBubbleBrickStageTest.sayForBubbleBrickStageTest
org.catrobat.catroid.uiespresso.content.brick.stage.ThinkForBubbleBrickStageTest.thinkForBubbleBrickStageTest
`
I already tested these Tests on my device, they worked fine without errors? why here are broken?

**Comment by ndrnour** [comment]
test this please

**Comment by ndrnour** [comment]
please run espresso tests

**Comment by ndrnour** [comment]
please run espresso tests

**Comment by ndrnour** [comment]
test this please

**Comment by ndrnour** [comment]
Pics
screenshot_20180227-153043
screenshot_20180227-153028

**Comment by ndrnour** [comment]
test this please

**Comment by ndrnour** [comment]
please run espresso tests

**Comment by ndrnour** [comment]
please run espresso tests

**Comment by ndrnour** [comment]
please run espresso tests

**Comment by thmq** [comment]
test this please

**Comment by thmq** [comment]
please run espresso tests

**Comment by thmq** [comment]
@ndrnour could you please rebase and reopen, when you're done?

**Pull request closed without merging**